jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java

```diff
@@ -375,10 +378,13 @@ private Authorization refreshBearerAuth(@Nullable String wwwAuthenticate)
       }
     }
 
-    throw new RegistryAuthenticationFailedException(
-        registry,
-        repository,
-        "server did not return 'WWW-Authenticate: Bearer' header: " + wwwAuthenticate);
+    eventHandlers.dispatch(
+        LogEvent.debug(
+            "server did not return 'WWW-Authenticate: Bearer' header. Actual: " + wwwAuthenticate));
+    if (readOnlyBearerAuth) {
+      return Verify.verifyNotNull(initialBearerAuthenticator.get()).authenticatePull(credential);
+    }
+    return Verify.verifyNotNull(initialBearerAuthenticator.get()).authenticatePush(credential);
```

Comment:
Would it be worth it to pull this out into a function to reduce duplication between here and L374-377?

Reply:
I think this is fine, considering that there's a difference between `authenticator` and `initialBearerAuthenticator` that the latter is nullable.